pruner/pruner.go

```diff
@@ -10,4 +10,5 @@ import (
 // from the node's datastore.
 type Pruner interface {
 	Prune(context.Context, *header.ExtendedHeader) error
+	Kind() string
```

Comment:
Okay so i hate this

we can implicitly do this by checking for the presence of a historical block (without its .q4 file) but I think that could be much more invasive and annoying to sort out.

Reply:
I think this is the cleanest option as of right now.

Reply:
Meh, I don't like the leaking abstraction. Did you consider persisting implementation identifier inside Pruner implementation? Perhaps it can be part to implementation to detect unwanted migration